### PR TITLE
Code cleanup + Pagination

### PR DESCRIPTION
**Pull request by samimahmood**
- [ ] Pagination for spotlights client side
- [ ] Filters and pagination for films and scenes submission
- [ ] Film Submission Pro Checkbox
- [ ] Film submission is showing in films without accepting it: FALSE (it is working as intended)
- [ ] In the future, maybe we want to add some validations on signup to prevent spam.

### CODE REVIEW

File: routes/Client/index.js [hunk 1]
```diff
@@ -3,16 +3,21 @@ var router = express.Router();
...
+    // var spamAcc = [];
+    // var users = await database.getUserList();
+    // users.forEach( u =>{
+    //     if (u.first_name.length > 25 || u.last_name.length > 25 && !u.subscription) {
+    //         spamAcc.push(u);
+    //     }
+    // });
+    // var userExcel = new UserExcelService();
+    // userExcel.allUsersExcelSheet(spamAcc, "Spam");
+    // res.json(spamAcc);
```
Comment (hashimhayat) [inline routes/Client/index.js]: Let's remove commented out code

File: routes/Admin/submissions.js [hunk 2]
```diff
@@ -10,7 +10,17 @@ router.get('/films', function (req, res, next) {
...
+                res.render('Admin/films/submissions', {
+                    layout: 'adminLayout',
+                    rows: result.docs,
+                    pagination: result.pagination,
+                    currentPage: result.page,
+                    nextPage: parseInt(result.page) +1,
+                    prevPage: parseInt(result.page) -1,
+                    totalPages: result.pages,
+                    sortQuery: result.sortQuery,
+                    statusQuery: result.statusQuery,
```
Comment (hashimhayat) [inline routes/Admin/submissions.js]: As discussed, pagination should be one passed to the front end. Everything related to pagination should be encapsulated in that object.

File: routes/Client/spotlights.js [hunk 3]
```diff
@@ -2,17 +2,21 @@ var express = require('express');
...
+            result => {
+                this.paginationObject = new PaginationModel(result);
+                this.paginationObject.toJson();
                 res.render('Client/spotlights/spotlights', {
                     title: `Spotlights | Labocine`,
                     BASE_URL: process.env.BASE_URL,
                     appStore: true,
-                    spotlights: result.filter(it => it.is_published == true),
+                    spotlights: paginationObject.docs,
+                    paginationObject: paginationObject,
```
Comment (hashimhayat) [inline routes/Client/spotlights.js]: Why don't we just call it pagination?

File: src/models/helper/PaginationModel.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+class PaginationModel {
```
Comment (hashimhayat) [inline src/models/helper/PaginationModel.js]: Let's call it Pagination. Also, it's not a model so it should not be in models/helper, rather in utils or somewhere else.

File: src/models/helper/PaginationModel.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+        this.paginationResult = paginationResult;
+        this.pagination = [];
+        for (var i = 1; i <= this.paginationResult.pages; i++) {
+            this.pagination.push({
+                id: i,
+                currentPage: this.paginationResult.page,
+                totalPages: this.paginationResult.pages,
+            })
+        }
```
Comment (hashimhayat) [inline src/models/helper/PaginationModel.js]: What's happening here?

File: routes/Admin/submissions.js (hunk 2 above)
Comment (samimahmood) [reply routes/Admin/submissions.js]: So i did it only at one place now so If there is a change I don't have to change at all the places, I will do it once it get perfect

File: src/models/helper/PaginationModel.js (hunk 5 above)
Comment (samimahmood) [reply src/models/helper/PaginationModel.js]: So we have to pass an array to the forntend so it knows how mange pages to show at the bottom pagination bar, so each array object have the information needed like page number and stuff

File: src/utils/Pagination.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+class Pagination {
+    constructor(dbResult) {
+
+        this.dbResult = dbResult;
```
Comment (hashimhayat) [inline src/utils/Pagination.js]: Let's not keep it. Rather take out all the important stuff from this object.

File: src/utils/Pagination.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    getPaginationPages(){
+        var paginationPages = [];
+        for (var i = 1; i <= this.dbResult.pages; i++) {
+            paginationPages.push({
+                id: i,
+                currentPage: this.dbResult.page,
+                totalPages: this.dbResult.pages,
+            })
+        }
+        return paginationPages;
```
Comment (hashimhayat) [inline src/utils/Pagination.js]: Why not use map?

File: routes/Client/spotlights.js [hunk 8]
```diff
@@ -2,17 +2,21 @@ var express = require('express');
...
 
 router.get('/', async function (req, res, next) {
     var accessPlaylist = await isProOrActiveSubscription(req);
-    database.getSpotlightList()
+    database.getSpotlightListPaginatedClient('-created_on', req.query.page)
         .then(
-             result => {
+            result => {
+                this.pagination = new Pagination(result);
+                this.pagination.toJson();
```
Comment (hashimhayat) [inline routes/Client/spotlights.js]: Do we need this?

Comment (hashimhayat) [reply routes/Client/spotlights.js]: Please do some research on how classes behave in JS as compared to JSON objects.

File: src/controllers/External/FilmSubmissionController.js [hunk 9]
```diff
@@ -64,17 +71,24 @@ class FilmSubmissionController extends Controller {
 
     processGet() {
         return new Promise((resolve, reject) => {
-            database.getFilmSubmissionsList().then(
+            if (this.statusQuery == "PENDING") {
+                this.statusQuery = null
+            }
```
Comment (hashimhayat) [inline src/controllers/External/FilmSubmissionController.js]: Can we improve this?

File: src/controllers/External/FilmSubmissionController.js [hunk 10]
```diff
@@ -64,17 +71,24 @@ class FilmSubmissionController extends Controller {
...
+            database.getFilmSubmissionsListPaginated(this.sortQuery, 'status', this.statusQuery, this.pageQuery).then(
                 result => {
-                    result.map(it => {
+                    result.docs.map(it => {
                         if (it.status === undefined) {
```
Comment (hashimhayat) [inline src/controllers/External/FilmSubmissionController.js]: So if someone submits a film or scene, the status should be pending not undefined.

File: src/controllers/External/FilmSubmissionController.js [hunk 11]
```diff
@@ -64,17 +71,24 @@ class FilmSubmissionController extends Controller {
...
                             it.status = "PENDING";
                         }
                         it.createdOn = Date(it.created_on).split(' ').slice(1, 4).join(' ')
                     });
+                    this.pagination = new Pagination(result);
+                    result.pagination = this.pagination.toJson();
+                    result.sortQuery = this.sortQuery.toLowerCase();
+                    result.statusQuery = this.statusQuery.toLowerCase();
```
Comment (hashimhayat) [inline src/controllers/External/FilmSubmissionController.js]: Maybe move this to the router?

File: src/controllers/External/FilmSubmissionController.js [hunk 12]
```diff
@@ -64,17 +71,24 @@ class FilmSubmissionController extends Controller {
...
+            }
+            database.getFilmSubmissionsListPaginated(this.sortQuery, 'status', this.statusQuery, this.pageQuery).then(
                 result => {
-                    result.map(it => {
+                    result.docs.map(it => {
                         if (it.status === undefined) {
                             it.status = "PENDING";
                         }
                         it.createdOn = Date(it.created_on).split(' ').slice(1, 4).join(' ')
                     });
```
Comment (hashimhayat) [inline src/controllers/External/FilmSubmissionController.js]: Figure this out why we have to do this? Seems like there is a bug somewhere.

File: src/models/FilmSubmissionModel.js [hunk 13]
```diff
@@ -38,6 +42,12 @@ const FilmsSubmissionModel = Schema({
     updated_on: {type:Number, default: null},
 }, {collection: 'films_submissions'} );
 
+FilmsSubmissionModel.index({'$**': 'text'});
+
+FilmsSubmissionModel.plugin(mongoosePaginate);
+
+FilmsSubmissionModel.plugin(uniqueValidator);
+
```
Comment (hashimhayat) [inline src/models/FilmSubmissionModel.js]: nit We can remove extra lines.

File: src/services/DatabaseService.js [hunk 14]
```diff
@@ -1270,6 +1270,20 @@ class DatabaseService {
             );
         });
     }
+    getFilmSubmissionsListPaginated(sort, filter, value=undefined, page) {
+        return new Promise((resolve, reject) => {
+            var query = {[filter]: value};
+            var options = {
+                limit: 100,
```
Comment (hashimhayat) [inline src/services/DatabaseService.js]: Why is the limit set to 100?

**Pull request merged**